src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -965,21 +979,14 @@ public void Connect(IPAddress[] addresses, int port)
                         lastex = null;
                         break;
                     }
-                    catch (Exception ex)
+                    catch (Exception ex) when (!ExceptionCheck.IsFatal(ex))
```

Comment:
I'd be fine deleting them all in a separate pass (IIRC at this point IsFatal really just checks for OOMs).  For now I'll keep it to avoid changing any of that logic in this PR; all I did here was take advantage of exception filters to neaten up the code slightly.